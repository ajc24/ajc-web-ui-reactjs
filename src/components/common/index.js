/**
 * Developed by Anthony Cox in 2024
 */
import { storybookCssClasses } from '../data/storybook';

/* Commonly used CSS classes */
const ajcBodyDefaultCss = 'ajc-body-default';

/**
 * Setup and configure the current web page to ensure the default CSS styling is set to it.
 * This also handles the styling of pages which are deployed on Storybook
 */
const setupDocumentBodyCss = () => {
  /* If we are rendering on Storybook, remove the Storybook classes which will affect the width of the component */
  if (document.body.classList.contains(storybookCssClasses.sbShowMain) === true) {
    document.body.classList.remove(storybookCssClasses.sbShowMain);
  }
  if (document.body.classList.contains(storybookCssClasses.sbMainPadded) === true) {
    document.body.classList.remove(storybookCssClasses.sbMainPadded);
  }

  /* Ensure that the <body> element stretches to 100% of the width of the screen, that the X axis scrollbar is hidden and Y axis scrollbar is always displayed */
  if (document.body.classList.contains(ajcBodyDefaultCss) === false) {
    document.body.classList.add(ajcBodyDefaultCss);
    document.activeElement.blur();
  }
};

export {
  setupDocumentBodyCss,
};
