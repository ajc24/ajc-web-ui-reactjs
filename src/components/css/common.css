/**
 * Developed by Anthony Cox in 2025
 */
/* .background-transparent {
  background-color: transparent;
} */

.background-green {
  background-color: #1F862F;
}

.background-grey {
  background-color: #F6F6F6;
}

.background-grey-2 {
  background-color: #757575;
}

.background-red {
  background-color: #EC0000;
}

.background-white {
  background-color: #FFFFFF;
}

.body-default {
  margin: 0px;
  padding: 0px;
  width: 100%;
  overflow-x: hidden;
  overflow-y: scroll;
}

/* .hidden {
  display: none;
}*/

.screen-width-content-inner {
  max-width: 1248px;
  min-width: 312px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

.screen-width-content-outer {
  max-width: 1920px;
  min-width: 344px;
  width: inherit;
}

.screen-width-root {
  width: 100%;
}
