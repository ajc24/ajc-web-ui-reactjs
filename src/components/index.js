/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
