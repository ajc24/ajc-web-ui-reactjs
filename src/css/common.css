/**
 * Developed by Anthony Cox in 2025
 */
.background-gold {
  background-color: #FFBF00;
}

.background-green {
  background-color: #006622;
}

.background-grey {
  background-color: #4D4D4D;
}

.background-grey-body {
  background-color: #F6F6F6;
}

.background-navy {
  background-color: #000066;
}

.background-red {
  background-color: #800000;
}

.background-transparent {
  background-color: transparent;
}

.background-white {
  background-color: #FFFFFF;
}

.body-default {
  margin: 0px;
  padding: 0px;
  width: 100%;
  overflow-x: hidden;
  overflow-y: scroll;
}

.common-component-width {
  max-width: 1248px;
  min-width: 328px;
}

.font-black {
  color: #000000;
}

.font-default {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 16px;
  word-wrap: break-word;
}

.font-gold {
  color: #FFBF00;
}

.font-white {
  color: #FFFFFF;
}

.hidden {
  display: none;
}

.screen-width-content-inner {
  max-width: 1248px;
  min-width: 328px;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

.screen-width-content-outer {
  max-width: 1920px;
  min-width: 360px;
  width: inherit;
}

.screen-width-root {
  width: 100%;
}