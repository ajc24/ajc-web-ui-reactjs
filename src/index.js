/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
