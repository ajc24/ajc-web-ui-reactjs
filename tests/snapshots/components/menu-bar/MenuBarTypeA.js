/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { MenuBarTypeA } from '../../../../src';
import { PageTemplateConfig } from '../../../../src/components/common';

describe('Menu Bar Type A', () => {
  describe('Menu Bar Type A with black background (default)', () => {
    let setupDocumentBodyCssSpy;
    let snapshot;
    const testComponentId = 'test-menu-bar-bg-default';

    beforeAll(() => {
      setupDocumentBodyCssSpy = jest
        .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
        .mockImplementation(() => {});
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MenuBarTypeA id={testComponentId}>
            Menu bar text content.
          </MenuBarTypeA>
        </React.Fragment>
      );
    });

    afterAll(() => {
      setupDocumentBodyCssSpy.mockRestore();
    });
    
    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Menu Bar Type A with grey background', () => {
    let setupDocumentBodyCssSpy;
    let snapshot;
    const testComponentId = 'test-menu-bar-bg-grey';

    beforeAll(() => {
      setupDocumentBodyCssSpy = jest
        .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
        .mockImplementation(() => {});
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MenuBarTypeA id={testComponentId} backgroundColour="grey">
            Menu bar text content.
          </MenuBarTypeA>
        </React.Fragment>
      );
    });

    afterAll(() => {
      setupDocumentBodyCssSpy.mockRestore();
    });
    
    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Menu Bar Type A with red background', () => {
    let setupDocumentBodyCssSpy;
    let snapshot;
    const testComponentId = 'test-menu-bar-bg-red';

    beforeAll(() => {
      setupDocumentBodyCssSpy = jest
        .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
        .mockImplementation(() => {});
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <MenuBarTypeA id={testComponentId} backgroundColour="red">
            Menu bar text content.
          </MenuBarTypeA>
        </React.Fragment>
      );
    });

    afterAll(() => {
      setupDocumentBodyCssSpy.mockRestore();
    });
    
    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
