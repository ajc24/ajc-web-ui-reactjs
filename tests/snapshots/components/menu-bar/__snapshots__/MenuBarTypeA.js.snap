// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Bar Type A Menu Bar Type A with black background (default) verifies the snapshot for the component 1`] = `"<div role="navigation" id="test-menu-bar-bg-default--container" class="ajc-container-screen ajc-background-black"><div class="ajc-container-screen-inner"><div id="test-menu-bar-bg-default--content" class="ajc-menu-bar-type-a-content ajc-font-default ajc-font-white">Menu bar text content.</div></div></div>"`;

exports[`Menu Bar Type A Menu Bar Type A with grey background verifies the snapshot for the component 1`] = `"<div role="navigation" id="test-menu-bar-bg-grey--container" class="ajc-container-screen ajc-background-grey-2"><div class="ajc-container-screen-inner"><div id="test-menu-bar-bg-grey--content" class="ajc-menu-bar-type-a-content ajc-font-default ajc-font-white">Menu bar text content.</div></div></div>"`;

exports[`Menu Bar Type A Menu Bar Type A with red background verifies the snapshot for the component 1`] = `"<div role="navigation" id="test-menu-bar-bg-red--container" class="ajc-container-screen ajc-background-red"><div class="ajc-container-screen-inner"><div id="test-menu-bar-bg-red--content" class="ajc-menu-bar-type-a-content ajc-font-default ajc-font-white">Menu bar text content.</div></div></div>"`;
